### PR TITLE
feat(collections): add reduceGroups

### PR DESCRIPTION
**Pull request by majidsajadi**

### CODE REVIEW

**Comment by majidsajadi** [comment]
@LionC Review please

File: collections/reduce_groups.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  reducer: (accumulator: A, current: T) => A,
+  initialValue: A,
+): Record<string, A> {
+  const result: Record<string, A> = {};
+
+  for (const [key, values] of Object.entries(record)) {
+    result[key] = values.reduce(reducer, initialValue);
+  }
+
+  return result;
```
[the remark this answers is not in the document]
Comment (majidsajadi) [reply collections/reduce_groups.ts]: @LionC Updated. let me know if you have anything else in mind.

**Review by LionC — APPROVED** [review]
LGTM, pingin @kt3k . Even though the implementation looks obvious, I think this is worth it, because if you want to do this cleanly, there is a lot of nesting that is distracting. But I will leave the decision to you :-)

**Review by kt3k — APPROVED** [review]
@majidsajadi LGTM. Thank you for your contribution!

@LionC I think this utility does enough complex operation, and has enough flexibility (you can provide any reducer). I agree with the inclusion! 👍

**Pull request merged**